### PR TITLE
Deprecate EVP_PKEY_set1_tls_encodedpoint()

### PR DESCRIPTION
**Pull request by mattcaswell**
We also deprecate EVP_PKEY_get1_tls_encodedpoint()

We replace them with new functions called EVP_PKEY_set1_encoded_public_key() and EVP_PKEY_get1_encoded_public_key(). The functionality remains the same but the name is much better. There is no reason for these functions to be TLS specific (and #13088 is using them in a CMS context), and they don't just encode points (they can also be used for FFDH).

### CODE REVIEW

**Comment by mattcaswell** [comment]
I see this as independent of #13088, and they can be merged in either order. However whichever one goes in second will require a small fixup to adjust for the changes in the other one.

**Comment by mattcaswell** [comment]
I get a weird doc-nits error with this. It complains about functions that are supposed to be internal but are documented as public. But I don't know why and I don't know how to fix it...

````
/usr/bin/perl ./util/find-doc-nits -n -l -e
EVP_PKEY_get1_tls_encodedpoint(3) is supposedly internal but is documented as public
EVP_PKEY_set1_tls_encodedpoint(3) is supposedly internal but is documented as public
make: *** [Makefile:3658: doc-nits] Error 1
````

File: crypto/evp/p_lib.c [hunk 1]
```diff
@@ -1316,30 +1316,32 @@ int EVP_PKEY_supports_digest_nid(EVP_PKEY *pkey, int nid)
...
 
-int EVP_PKEY_set1_tls_encodedpoint(EVP_PKEY *pkey,
-                               const unsigned char *pt, size_t ptlen)
+int EVP_PKEY_set1_encoded_public_key(EVP_PKEY *pkey, const unsigned char *pub,
+                                     size_t publen)
 {
     if (pkey->ameth == NULL) {
-        OSSL_PARAM params[2] = { OSSL_PARAM_END, OSSL_PARAM_END };
+        OSSL_PARAM params[2]
+            = { OSSL_PARAM_END, OSSL_PARAM_END };
```
Comment (levitte) [inline crypto/evp/p_lib.c:1324]: Nit: Why was this line break necessary?

Comment (mattcaswell) [reply crypto/evp/p_lib.c:1324]: It wasn't this is an artefact from an earlier iteration of development - so this change is no longer necessary and I have removed it.

**Comment by mattcaswell** [comment]
I rebased this to resolve a conflict with master, and pushed a fixup to address the nit raised by @levitte.

@levitte - I could use a hint on resolving the doc-nits failure. I have no idea why it is complaining.

**Comment by levitte** [comment]
Because `EVP_PKEY_get1_tls_encodedpoint` and `EVP_PKEY_set1_tls_encodedpoint` are macros, they need to be declared as such in util/other.syms.  If not present there, they are assumed to be "internal" (which is weird, I agree)

**Comment by mattcaswell** [comment]
Ping?

**Review by t8m — APPROVED** [review]
LGTM

**Comment by mattcaswell** [comment]
I had to make some changes now that #13088 has gone in. I've rebased and modified the CMS code accordingly.

@t8m please can you reconfirm?

**Review by t8m — APPROVED** [review]

**Comment by mattcaswell** [comment]
Pushed. Thanks!

**Pull request closed without merging**